feat: add support for `GUILD_AUDIT_LOG_ENTRY_CREATE` event (#915)
DisnakeDev/disnake: docs/api.rst

    :param payload: The raw event payload data.
    :type payload: :class:`RawIntegrationDeleteEvent`

Audit Logs
++++++++++

> .. function:: on_audit_log_entry_create(entry)

Reviewer: There should be a warning about the cache reliance here, since it's not receiving a full audit log object but only a single entry.
It therefore doesn't have access to all the [additional objects](https://discord.com/developers/docs/resources/audit-log#audit-log-object-audit-log-structure) usually provided by the API, so `AuditLogEntry.target` more frequently ends up as just an `Object` with the ID instead of a full webhook object or w/e

Author: What do you think about [this](527e9b946b70b91c8cd77cc87cb187fb961a2cff)?

Reviewer: Would be good to go a bit more into the cache aspect, "Discord limitation" sounds fairly vague. Seems good other than that :)

```rst
    .. warning::
        This scope of data in this gateway event is limited, which means it is much more
        reliant on the cache than :meth:`Guild.audit_logs`.
        Because of this, :attr:`AuditLogEntry.target` will frequently
        be of type :class:`Object` instead of the respective model.
```

Author: I knew I could count on you, can you suggest it, so I can commit it directly, please

Reviewer: done, thanks :pray:

    :param payload: The raw event payload data.
    :type payload: :class:`RawIntegrationDeleteEvent`

Audit Logs
++++++++++

.. function:: on_audit_log_entry_create(entry)

    Called when an audit log entry is created.
    You must have the :attr:`~Permissions.view_audit_log` permission to receive this.

    This requires :attr:`Intents.moderation` to be enabled.

>     .. warning::
>         Because of a Discord limitation, :attr:`AuditLogEntry.target` will frequently
>         be of type :class:`Object` instead of the respective model.

Reviewer: ```suggestion
    .. warning::
        This scope of data in this gateway event is limited, which means it is much more
        reliant on the cache than :meth:`Guild.audit_logs`.
        Because of this, :attr:`AuditLogEntry.target` will frequently
        be of type :class:`Object` instead of the respective model.
```

Author: thanks a lot

    :param payload: The raw event payload data.
    :type payload: :class:`RawIntegrationDeleteEvent`

Audit Logs
++++++++++

> .. function:: on_audit_log_entry_create(entry)

Reviewer: This is now missing from the new `Event` enum :sweat_smile:

Author: f46e769e120e62f9d4f16cdf28e8ea2639e0c6d1

.. class:: AuditLogAction

    Represents the type of action being done for a :class:`AuditLogEntry`\,

>     which is retrievable via :meth:`Guild.audit_logs` or via :func:`on_audit_log_entry_create` event.

Reviewer: ```suggestion
    which is retrievable via :meth:`Guild.audit_logs` or via the :func:`on_audit_log_entry_create` event.
```

        A member was kicked.

        When this is the action, the type of :attr:`~AuditLogEntry.target` is
        the :class:`User` who got kicked. If the user is not found then it is

>         a :class:`Object` with an ID.

Reviewer: ```suggestion
        a :class:`Object` with the user's ID.
```
(same thing for the other ones)

Author: b34735ec07eb783a861f52c349f0153b6bc2e121

>         a :class:`Object` with the user's ID.

Reviewer: A similar change is needed for `.message_delete`, `.message_pin`, `.message_unpin`, `.automod_block_message`, `.automod_send_alert_message` and `.automod_timeout`, all of which may have an `Object` target

Author: b8b241adc9923bab679daca945fb28bc22a18dc2
